tests/foreman/ui/test_gpgkey.py

```diff
@@ -384,7 +380,118 @@ def test_consume_content_1(self):
+        response = client.put(
+            entities.ContentView(id=content_view['id']).path(),
+            {u'repository_ids': [repo_id]},
+            auth=get_server_credentials(),
+            verify=False,
+        )
+        response.raise_for_status()
+        # step 6: Publish content view
+        entities.ContentView(id=content_view['id']).publish()
+        # step 6.2: Promote content view to lifecycle_env
+        content_view = entities.ContentView(id=content_view['id']).read_json()
+        self.assertEqual(len(content_view['versions']), 1)
+        entities.ContentViewVersion(
+            id=content_view['versions'][0]['id']
+        ).promote(lifecycle_env['id'])
+        # step 7: Create activation key
+        ak_id = entities.ActivationKey(
+            name=activation_key_name,
+            environment=lifecycle_env['id'],
+            organization=self.org_id,
+            content_view=content_view['id'],
+        ).create_json()['id']
+        for subscription in entities.Organization(
+                id=self.org_id).subscriptions():
+            if subscription['product_name'] == product_name:
+                entities.ActivationKey(id=ak_id).add_subscriptions({
+                    'quantity': 1,
+                    'subscription_id': subscription['id'],
+                })
+                break
+        # Create VM
+        package_name = "cow"
+        with VirtualMachine(distro='rhel66') as vm:
+            # Download and Install rpm
+            result = vm.run(
+                "wget -nd -r -l1 --no-parent -A '*.noarch.rpm' http://{0}/pub/"
+                .format(conf.properties['main.server.hostname'])
+            )
+            self.assertEqual(
+                result.return_code, 0,
+                "failed to fetch katello-ca rpm: {0}, return code: {1}"
+                .format(result.stderr, result.return_code)
+            )
+            result = vm.run(
+                'rpm -i katello-ca-consumer*.noarch.rpm'
+            )
+            self.assertEqual(
+                result.return_code, 0,
+                "failed to install katello-ca rpm: {0} and return code: {1}"
+                .format(result.stderr, result.return_code)
+            )
+            # Register client with foreman server using activation-key
+            result = vm.run(
+                u'subscription-manager register --activationkey {0} '
+                '--org {1} --force'
+                .format(activation_key_name, self.org_label)
+            )
+            '''self.assertEqual(
+                result.return_code, 0,
+                "failed to register client:: {0} and return code: {1}"
+                .format(result.stderr, result.return_code)
+            )'''
+            # Validate if gpgcheck flag is enabled in repo file
+            repo_file = "/etc/yum.repos.d/redhat.repo"
+            result = vm.run(
+                'cat {0} | grep gpgcheck | cut -d " " -f3'
+                .format(repo_file)
```

Comment:
Does this really work? `gpgcheck` often looks like this in a `.repo` file:

    gpgcheck=1

So using a delimiter of `" "` doesn't seem right. Also:

* You can grep a file directly. You can write `grep gpgcheck {0}` instead of `cat {0} | grep gpgcheck`.
* I find awk to be simpler than cut when splitting fields. For example, if printing out the second column in a whitespace-delimited piece of text, this works: `… | awk '{print $2}'`. Or if you wanted to use `/` as a delimiter: `… | awk -F / '{print $2}'`.